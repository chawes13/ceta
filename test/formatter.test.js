const { createMessage, createErrorMessage } = require('../src/formatter')
const exampleResponse = require('../fixtures/example-response.json')

describe('Formats messages correctly for Slack', () => {
  describe('Create Message', () => {
    test('Creates an ephemeral message', () => {
      const userInput = "chicago br // purple // n"
      const response = {}
      const message = createMessage(userInput, response)
      expect(message.response_type).toBe('ephemeral')
    })
    
    test('Includes a header with the user\'s original input', () => {
      const userInput = "chicago br // purple // n"
      const response = {}
      const message = createMessage(userInput, response)
      expect(message.blocks[0].text.text).toContain(userInput)
    })
    
    test('Creates a message matching Slack\'s Block API kit syntax', () => {
      const userInput = "fullerton // red"
      const response = exampleResponse
      
      const realDateNow = Date.now
      global.Date.now = jest.fn(() => new Date('March 4, 2019 03:24:13'))
      
      const message = createMessage(userInput, response)
      expect(message).toMatchSnapshot('slack-block-api')
      
      // restore mock
      global.Date.now = realDateNow
    })
    
    test('Provides the time that the estimates were generated', () => {
      const userInput = "chicago br // purple // n"
      const response = {}
      
      const realDateNow = Date.now
      global.Date.now = jest.fn(() => new Date('March 4, 2019 03:24:13'))
      
      const message = createMessage(userInput, response)
      expect(message.blocks[message.blocks.length - 1].elements[0].text).toContain('03:24:13')
      
      // restore mock
      global.Date.now = realDateNow
    })
  })
  
  describe('Create Error Message', () => {
    test('Creates an ephemeral message', () => {
      const error = new Error('uh oh')
      const message = createErrorMessage(error)
      expect(message.response_type).toBe('ephemeral')
    })
    
    test('Creates an error message with the error name and message', () => {
      const error = new Error('uh oh')
      const message = createErrorMessage(error)
      expect(message).toHaveProperty('blocks')
      expect(message.blocks[0].text.text).toContain('Error: uh oh')
    })
  })
})
