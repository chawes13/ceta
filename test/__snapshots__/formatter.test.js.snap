// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Formats messages correctly for Slack Create Message Creates a message matching Slack's Block API kit syntax: slack-block-api 1`] = `
Object {
  "blocks": Array [
    Object {
      "text": Object {
        "text": "Arrival Estimates for: \\"*fullerton // red*\\"",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "text": Object {
        "text": "*Service toward Howard* 
 :red_circle:",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "text": Object {
        "text": "7 minutes (Red)
21 minutes (Red)",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "text": Object {
        "text": "*Service toward 95th/Dan Ryan* 
 :red_circle:",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "text": Object {
        "text": "DUE (Red)
9 minutes (Red)
15 minutes (Red)",
        "type": "mrkdwn",
      },
      "type": "section",
    },
    Object {
      "type": "divider",
    },
    Object {
      "elements": Array [
        Object {
          "text": ":stopwatch: Estimates generated at 03:24:13",
          "type": "mrkdwn",
        },
      ],
      "type": "context",
    },
  ],
  "response_type": "ephemeral",
  "text": "C(E)TA Arrival Estimates",
}
`;
